src/bullet.rs

```diff
@@ -11,13 +9,33 @@ impl Plugin for BulletPlugin {
     }
 }
 
-pub fn bullet_damage(mut enemies: Query<&mut Health>, mut events: EventReader<CollisionEvent>) {
+pub fn bullet_damage(
+    mut entities: Query<(&mut Health, &ChickenOrDog)>,
+    bullets: Query<(&Collisions, &Bullet)>,
+) {
+    // have to collect the iter to drop the entities query
+    #[allow(clippy::needless_collect)]
+    let entities_to_damage = bullets
+        .iter()
+        .flat_map(|(collisions, bullet)| {
+            collisions
+                .entities()
+                .map(move |collision| (collision, bullet))
```

Comment:
the logic here feels overly complex for the task being done

Reply:
the logic here is the check for bullet's origin team and the entities team. the flat map gets us to a list of entities and the bullet that collided.

Reply:
I see that, but differences in variable names of just the character s are often confusing,  it's fine here